src/main/java/nl/knaw/dans/vaultingest/core/rdabag/output/ZipBagOutputWriterFactory.java

```diff
@@ -30,15 +30,17 @@ public ZipBagOutputWriterFactory(Path outputDir) {
 
     @Override
     public BagOutputWriter createBagOutputWriter(Deposit deposit) throws IOException {
-        var doi = Objects.requireNonNull(deposit.getDoi(), "Deposit DOI is null");
-        // TODO version should be coming from the deposit, or should it?
-        // this is to be determined by Jan and Linda
-        var output = outputDir.resolve(outputFilename(doi, "1.0"));
+        var output = outputDir.resolve(outputFilename(deposit.getBagId(), deposit.getObjectVersion()));
         return new ZipBagOutputWriter(output);
     }
 
-    private Path outputFilename(String doi, String version) {
-        doi = doi.replaceAll("[^a-zA-Z0-9]", "-");
-        return Path.of(String.format("%s-v%s.zip", doi, version).toLowerCase());
+    String outputFilename(String bagId, Long objectVersion) {
+        Objects.requireNonNull(bagId);
+        Objects.requireNonNull(objectVersion);
+
+        // strip anything before the colon (if present), and also the colon itself
+        bagId = bagId.toLowerCase().replaceAll(".*:", "");
```

Comment:
Is there at most one colon? Otherwise you'll need `[^:]+:`
Oh, from the test I see you want to strip up to the *last* colon, so please adjust the comment.